types/src/v0/impls/stake_table.rs

```diff
@@ -439,15 +439,23 @@ impl StakeTableFetcher {
         to_block: u64,
     ) -> anyhow::Result<Vec<(EventKey, StakeTableEvent)>> {
         let persistence_lock = self.persistence.lock().await;
-        let res = persistence_lock.load_events().await?;
+        let (queried_l1, persistence_events) = persistence_lock.load_events(to_block).await?;
         drop(persistence_lock);
 
-        let from_block = res
-            .as_ref()
-            .map(|(block, _)| block + 1)
-            .filter(|from| *from <= to_block); // Only use from_block if it's less than to_block
+        tracing::info!("loaded events from storage to_block={to_block:?}");
 
-        tracing::info!("loaded events from storage from_block={from_block:?}");
+        // No need to fetch from contract
+        // if persistence returns all the events that we need
+        if queried_l1 == Some(to_block) {
```

Comment:
Should this be `>=` ?

Reply:
I guess not because we already trim the block number in `load_events`. This confused me a bit. Especially since the variable here is called `queried_l1`. I think what we care about are 1) if we had all the results in storage and 2) if not up to what l1 block we previously fetched.

Maybe a type like

```rust
enum EventsPersistenceRead {
    Complete,
    UntilL1Block(u64)
}
```

would make this more clear?